janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/StaticArrayEntryList.java

```diff
@@ -398,11 +401,89 @@ private static <E,D> EntryList of(Iterable<E> elements, StaticArrayEntry.GetColV
     private static <E,D> EntryList of(Iterator<E> elements, StaticArrayEntry.GetColVal<E,D> getter, StaticArrayEntry.DataHandler<D> dataHandler) {
         Preconditions.checkArgument(elements!=null && getter!=null && dataHandler!=null);
         if (!elements.hasNext()) return EMPTY_LIST;
+
+        EntryListComputationContext context = generateComputationContext();
+        applyElementsComputation(elements, getter, dataHandler, context);
+        return convert(context);
+    }
+
+    public static EntryListComputationContext generateComputationContext(){
         long[] limitAndValuePos = new long[10];
-        byte[] data = new byte[limitAndValuePos.length*15];
-        EntryMetaData[] metadataSchema = null;
-        int pos=0;
-        int offset=0;
+        return new EntryListComputationContext(limitAndValuePos, new byte[limitAndValuePos.length*15], null, 0, 0);
+    }
+
+    public static <E> void supplyEntryList(ChunkedJobDefinition<Iterator<E>, EntryListComputationContext, EntryList> chunkedJobDefinition,
+                                           StaticArrayEntry.GetColVal<E,StaticBuffer> getter,
+                                           ExecutorService executorService) {
+        supplyEntryList(chunkedJobDefinition, getter, StaticArrayEntry.StaticBufferHandler.INSTANCE, executorService);
+    }
+
+    private static <E,D> void supplyEntryList(ChunkedJobDefinition<Iterator<E>, EntryListComputationContext, EntryList> chunkedJobDefinition,
+                                              StaticArrayEntry.GetColVal<E,D> getter,
+                                              StaticArrayEntry.DataHandler<D> dataHandler,
+                                              ExecutorService executorService){
+        assert chunkedJobDefinition !=null && getter!=null && dataHandler!=null;
+
+        executorService.execute(() -> {
+
+            Queue<Iterator<E>> chunksQueue = chunkedJobDefinition.getDataChunks();
+
+            if(!chunkedJobDefinition.getProcessingLock().tryLock()){
+                return;
+            }
+
+            try {
+                if(chunkedJobDefinition.getResult().isDone()){
+                    return;
```

Comment:
Do we not need to release the lock?

Reply:
It is released in `finally` block below (`chunkedJobDefinition.getProcessingLock().unlock();`).
In Java `finally` block is executed no-matter what `try` / `catch` blocks produced.

Reply:
Missed that, thanks!